Main.hs

```diff
@@ -14,7 +14,7 @@ expr (N v) = N v
 expr (S v) = S v
 expr (List v) = List v
 expr (Error v) = S v
-expr (If cond exprl exprr) = ifExpr cond exprl exprr
+expr (If cond exprThen exprElse) = ifExpr cond exprThen exprElse
```

Comment:
What about the case when the `else` expression is optional?

Reply:
![image](https://user-images.githubusercontent.com/14208546/75513574-1af81600-59aa-11ea-9065-c8f907f22efd.png)

Reply:
@devsebb said earlier in the planning phase that we can just pass `Nil` into the else clause.

Reply:
We'll need to add `Nil` to the `Expr` grammar in order to use it, though.

Reply:
Let's require always have an `else` expression. no non-else if expression. then, if user doesn't want to do anything, just put empty ()